services/cert/auto_certificate.go

```diff
@@ -0,0 +1,177 @@
+	c.modifyConfigURL(location)
+	autoCertificates, apiResponse, err := c.sdkClient.AutoCertificateApi.AutoCertificatesGet(ctx).Execute()
+	apiResponse.LogInfo()
+	return autoCertificates, apiResponse, err
+}
+
+//nolint:golint
+func (c *Client) CreateAutoCertificate(ctx context.Context, location string, autoCertificatePostData certmanager.AutoCertificateCreate) (certmanager.AutoCertificateRead, *certmanager.APIResponse, error) {
+	c.modifyConfigURL(location)
+	autoCertificate, apiResponse, err := c.sdkClient.AutoCertificateApi.AutoCertificatesPost(ctx).AutoCertificateCreate(autoCertificatePostData).Execute()
+	apiResponse.LogInfo()
+	return autoCertificate, apiResponse, err
+}
+
+//nolint:golint
+func (c *Client) UpdateAutoCertificate(ctx context.Context, autoCertificateID, location string, autoCertificatePatchData certmanager.AutoCertificatePatch) (certmanager.AutoCertificateRead, *certmanager.APIResponse, error) {
+	c.modifyConfigURL(location)
+	autoCertificate, apiResponse, err := c.sdkClient.AutoCertificateApi.AutoCertificatesPatch(ctx, autoCertificateID).AutoCertificatePatch(autoCertificatePatchData).Execute()
+	apiResponse.LogInfo()
+	return autoCertificate, apiResponse, err
+}
+
+//nolint:golint
+func (c *Client) DeleteAutoCertificate(ctx context.Context, autoCertificateID, location string) (*certmanager.APIResponse, error) {
+	c.modifyConfigURL(location)
+	apiResponse, err := c.sdkClient.AutoCertificateApi.AutoCertificatesDelete(ctx, autoCertificateID).Execute()
+	apiResponse.LogInfo()
+	return apiResponse, err
+}
+
+//nolint:golint
+func (c *Client) IsAutoCertificateReady(ctx context.Context, d *schema.ResourceData) (bool, error) {
+	autoCertificateID := d.Id()
+	location := d.Get("location").(string)
+	autoCertificate, _, err := c.GetAutoCertificate(ctx, autoCertificateID, location)
+	if err != nil {
+		return false, fmt.Errorf("error checking auto-certificate status: %w", err)
+	}
+	if autoCertificate.Metadata == nil || autoCertificate.Metadata.State == nil {
+		return false, fmt.Errorf("metadata or state is empty for auto-certificate with ID: %v", autoCertificateID)
+	}
+	if utils.IsStateFailed(*autoCertificate.Metadata.State) {
+		return false, fmt.Errorf("error while checking if auto-certificate is ready, auto-certificate ID: %v, state: %v", autoCertificateID, *autoCertificate.Metadata.State)
+	}
+	return strings.EqualFold(*autoCertificate.Metadata.State, constant.Available), nil
+}
+
+//nolint:golint
+func (c *Client) IsAutoCertificateDeleted(ctx context.Context, d *schema.ResourceData) (bool, error) {
+	autoCertificateID := d.Id()
+	location := d.Get("location").(string)
+	autoCertificate, apiResponse, err := c.GetAutoCertificate(ctx, autoCertificateID, location)
+	if err != nil {
+		if apiResponse.HttpNotFound() {
+			return true, nil
+		}
+		return false, fmt.Errorf("error while checking deletion status for auto-certificate with ID: %v, error: %w", d.Id(), err)
+	}
+	if utils.IsStateFailed(*autoCertificate.Metadata.State) {
```

Comment:
is a nilcheck required here before the derefernce?

Reply:
@cbeti-ionos done in ec747230649ad96067a92e1d3b11d469d3eda2f1